tests/support/kubernetes.py

```diff
@@ -871,6 +1218,126 @@ async def read_namespaced_pod_status(
         self._maybe_error("read_namespaced_pod_status", name, namespace)
         return self._get_object(namespace, "Pod", name)
 
+    # REPLICASET API
+
+    async def create_namespaced_replica_set(
+        self, namespace: str, body: V1ReplicaSet
+    ) -> None:
+        """Create a ``ReplicaSet`` object.
+
+        This mock object ignores its ``replicas`` and ReplicaSpec field
+        and just unconditionally creates one pod (with the same name).  Then
+        set that pod to running.
+
+        Parameters
+        ----------
+        namespace
+            Namespace in which to store the object.
+        body
+            Object to store.
+
+        Raises
+        ------
+        ApiException
+            Raised with 409 status if the object already exists.
+        """
+        self._maybe_error("create_namespaced_replica_set", namespace, body)
+        name = body.metadata.name
+        await self.create_namespaced_pod(
+            namespace,
+            V1Pod(
+                metadata=V1ObjectMeta(name=name),
+            ),
+        )
+        pod = await self.read_namespaced_pod(name, namespace)
+        pod.status.phase = "Running"
+        self._update_metadata(pod, "v1", "Pod", namespace)
+        self._update_metadata(body, "v1", "ReplicaSet", namespace)
+        self._store_object(namespace, "ReplicaSet", name, body)
```

Comment:
As above, this should be done before creating the pod.

Reply:
Replicaset/Deployment are gone